src/core/thread/mesh_forwarder.cpp

```diff
@@ -1505,12 +1505,20 @@ void MeshForwarder::HandleFragment(FrameData            &aFrameData,
             // Security Check: only consider reassembly buffers that had the same Security Enabled setting.
             if (msg.GetLength() == fragmentHeader.GetDatagramSize() &&
                 msg.GetDatagramTag() == fragmentHeader.GetDatagramTag() &&
-                msg.GetOffset() == fragmentHeader.GetDatagramOffset() &&
+                msg.GetOffset() <= fragmentHeader.GetDatagramOffset() &&
```

Comment:
MAC Frame Security should drop duplicate MAC frames. Can you clarify when a given 6LoWPAN fragment would not be dropped by duplicate frame detection?

Reply:
the root cause for this issue is identified as a radio bug that was using a new sequence number for frame retransmission. This is a defect prevention fix, in case there are bugs in radio hardware, to avoid fragment drops.

Reply:
Thanks, @arun-mahasenan.

I'd prefer not to include this change in `MeshForwarder`. While I understand the goal of protecting against possible issues in the `Mac` or radio platform, this change might mask those issues instead. The code here only applies when the frame has a fragment header and wouldn't provide protection in other cases (mesh header frames, regular lowpan frames, etc.).

We expect the MAC/radio layer to detect and filter duplicate received frames before passing them to the next layer (`MeshForwarder`). This way it applies to all types of frames.